ocaml/xapi/context.ml

```diff
@@ -229,6 +229,34 @@ let parent_of_origin (origin : origin) span_name =
 
 let attribute_helper_fn f v = Option.fold ~none:[] ~some:f v
 
+let addr_port_of_sock s =
+  match s with
+  | Unix.ADDR_UNIX "" ->
+      (None, None)
+  | Unix.ADDR_UNIX soket_name ->
+      (Some soket_name, None)
+  | Unix.ADDR_INET (addr, port) ->
+      (Some (Unix.string_of_inet_addr addr), Some (string_of_int port))
+
+let attr_of_fd s =
+  let peer_addr = s |> Unix.getpeername |> addr_port_of_sock in
+  let local_addr = s |> Unix.getsockname |> addr_port_of_sock in
+  [
```

Comment:
how about 
```
List.combine ["network.local.address"; "network.local.port"; "network.peer.address"; "network.peer.port"] [local_addr; local_port; peer_addr; peer_port] 
|> List.filter (fun (_n, n) -> Option.is_some n) 
```

Reply:
At that point I don't think the List.combine is needed, simply define the list of tuples and filter it